### ISSUE TITLE
TGI inference results are different from those by native transformer inference API

### ISSUE THREAD

**Issue by zengyijie** [huggingface/text-generation-inference#855]
### System Info

Some TGI inference results (either by docker or local install) are different from those by native transformer inference API, about 50% results are different, while the other 50% results are identical.
text-generation-inference local install info:
Target: x86_64-unknown-linux-gnu
Cargo version: 1.70.0
Commit sha: 7766fee9b15b6f600ee73db7f8bd72f7c62f335e
OS: Ubuntu 18.04.5 LTS, kernel: 4.15.0-175
CUDA: 11.8
GPU: 4 X Nvidia A100 40G, or 8 X 3090
Model: falcon 40B ([https://huggingface.co/tiiuae/falcon-40b])

### Information

- [X] Docker
- [X] The CLI directly

### Tasks

- [X] An officially supported command
- [ ] My own modifications

### Reproduction

1. Download falcon-40B model;
2. Install text-generation-inference
3. Start TGI docker image, and use curl command

docker run -d 
  --name test-falcon 
  --gpus='"device=0,1,6,7"' 
  -p 19999:80 
  -v /model/falcon/falcon-40b-instruct:/model/falcon-40b-instruct 
  ghcr.io/huggingface/text-generation-inference:sha-37df6df 
  --model-id /model/falcon-40b-instruct 
  --num-shard 4 
  --dtype bfloat16 
  --trust-remote-code 
  --max-total-tokens 4096

curl 127.0.0.1:19999/generate \
  -X POST \
  -d '{"inputs":"Can you write an article about what is beauty?","parameters":{"max_new_tokens":2048, "do_sample": true, "top_p": 0.5, "temperature": 0.8, "repetition_penalty": 1.2, "seed": 42}}' \
  -H 'Content-Type: application/json'

===========
Outputs:

Beauty can be defined as the quality or combination of qualities in a person, object, place, idea, or experience that gives pleasure to the senses and deep satisfaction to the mind. It encompasses everything from physical appearance to inner character traits such as kindness, intelligence, and creativity. Beauty is subjective and varies depending on individual preferences and cultural norms. Some may find natural landscapes beautiful while others prefer man-made structures. Similarly, some people value external appearances more than internal characteristics, while others prioritize personality over looks. Ultimately, beauty is something that resonates with us at a personal level and makes our lives richer and more fulfilling.

4. Use native transformer inference codes:

from transformers import AutoTokenizer, AutoModelForCausalLM, AutoConfig
import transformers
import torch

model_path = "/mnt/work/model/falcon/falcon-40b-instruct"

config = AutoConfig.from_pretrained(model_path, use_cache=True, trust_remote_code=True)
tokenizer = AutoTokenizer.from_pretrained(model_path)
model = AutoModelForCausalLM.from_pretrained(model_path, config=config, torch_dtype=torch.bfloat16, device_map="auto", trust_remote_code=True)
model.eval()

inputs = "Can you write an article about what is beauty?"

input_ids = tokenizer(inputs)['input_ids']
input_ids = torch.LongTensor([input_ids]).to("cuda")

output_ids = model.generate(input_ids, 
                            max_new_tokens=2048,
                            do_sample=True, 
                            temperature=0.8, 
                            top_p=0.5, 
                            num_return_sequences=1, 
                            repetition_penalty=1.2)[0]
output = tokenizer.decode(output_ids)

print("outputs:")
print(output)

=============
Outputs:
Beauty can be defined as the quality or combination of qualities in a person, object, place, idea, or experience that gives pleasure to the senses and deep satisfaction to the mind. It encompasses everything from physical attributes like symmetry and proportion to intangible traits such as kindness and intelligence. Beauty is subjective and varies depending on individual perception and cultural context.

### Expected behavior

The inference results of TGI and native transformer API should most be the same.

**Comment by Narsil** [comment]
Tiny differences for different implementation is actually not that all surprising.
Tiny drifts can occur in many places, different hardware, different kernels, different torch versions.

TGI uses different kernels than `transformers`. Therefore the late drift is actually perfectly OK.

If you can see catastrophic failure in one case or the other, that would be concerning.
Closing for now.
